File: app/src/main/java/com/example/whattowatch/api/IMovieAPI.java
package com.example.whattowatch.api;

import com.example.whattowatch.model.response.cradits.MovieCredits;
import com.example.whattowatch.model.response.detail.DetailMovie;
import com.example.whattowatch.model.response.list.MoviesObject;
import com.example.whattowatch.model.response.person.Person;
import com.example.whattowatch.model.response.video.MovieVideosResponse;

import io.reactivex.Flowable;
import retrofit2.http.GET;
import retrofit2.http.Path;
import retrofit2.http.Query;

public interface IMovieAPI {


     String BASE_PICTURE = "https://image.tmdb.org/t/p/w500/";

    @GET("movie/{section}")
    Flowable<MoviesObject> getMovies(@Path("section") String section,
                                     @Query("language") String language,
                                     @Query("page") int page);

    @GET("movie/{section}") //для отображения Upcoming нужно указать регион
    Flowable<MoviesObject> getUpcommingMovies(@Path("section") String section,
                                              @Query("language") String language,
                                              @Query("page") int page,
                                              @Query("region") String region);

    @GET("movie/{movie_id}")
    Flowable<DetailMovie> getDetail(@Path("movie_id") int id,
                                      @Query("language") String language);

    @GET("movie/{id}/videos")
    Flowable<MovieVideosResponse> getMovieVideos(@Path("id") Integer id);

    @GET("search/movie")
    Flowable<MoviesObject> searchMovie(@Query("language") String language,
                                       @Query("page") int page,
                                       @Query("query") String query);

    @GET("movie/{id}/credits")
    Flowable<MovieCredits> getMovieCredits(@Path("id") Integer id);

    @GET("person/{id}")
    Flowable<Person> getPersonDetail(@Path("id") int id,
                                     @Query("language") String language);


}

File: app/src/main/java/com/example/whattowatch/api/IMovieAPI.kt
package com.example.whattowatch.api

import com.example.whattowatch.model.response.cradits.MovieCredits
import com.example.whattowatch.model.response.detail.DetailMovie
import com.example.whattowatch.model.response.list.MoviesObject
import com.example.whattowatch.model.response.person.Person
import com.example.whattowatch.model.response.video.MovieVideosResponse
import io.reactivex.Flowable
import retrofit2.http.GET
import retrofit2.http.Path
import retrofit2.http.Query

interface IMovieAPI {
    @GET("movie/{section}")
    fun getMovies(@Path("section") section: String?,
                  @Query("language") language: String?,
                  @Query("page") page: Int): Flowable<MoviesObject>

    @GET("movie/{section}")
    fun  //для отображения Upcoming нужно указать регион
            getUpcommingMovies(@Path("section") section: String?,
                               @Query("language") language: String?,
                               @Query("page") page: Int,
                               @Query("region") region: String?): Flowable<MoviesObject>

    @GET("movie/{movie_id}")
    fun getDetail(@Path("movie_id") id: Int,
                  @Query("language") language: String?): Flowable<DetailMovie>

    @GET("movie/{id}/videos")
    fun getMovieVideos(@Path("id") id: Int?): Flowable<MovieVideosResponse>

    @GET("search/movie")
    fun searchMovie(@Query("language") language: String?,
                    @Query("page") page: Int,
                    @Query("query") query: String?): Flowable<MoviesObject>

    @GET("movie/{id}/credits")
    fun getMovieCredits(@Path("id") id: Int?): Flowable<MovieCredits>

    @GET("person/{id}")
    fun getPersonDetail(@Path("id") id: Int,
                        @Query("language") language: String?): Flowable<Person>

    companion object {
        const val BASE_PICTURE = "https://image.tmdb.org/t/p/w500/"
    }
}